tests/integration/test_system_ddl_worker_queue/test.py

```diff
@@ -25,26 +26,39 @@ def started_cluster():
     try:
         cluster.start()
 
-        for i, node in enumerate([node1, node2]):
-            node.query("CREATE DATABASE testdb")
-            node.query(
-                """CREATE TABLE testdb.test_table(id UInt32, val String) ENGINE = ReplicatedMergeTree('/clickhouse/test/test_table1', '{}') ORDER BY id;""".format(
-                    i
-                )
-            )
-        for i, node in enumerate([node3, node4]):
-            node.query("CREATE DATABASE testdb")
-            node.query(
-                """CREATE TABLE testdb.test_table(id UInt32, val String) ENGINE = ReplicatedMergeTree('/clickhouse/test/test_table2', '{}') ORDER BY id;""".format(
-                    i
-                )
-            )
         yield cluster
 
     finally:
         cluster.shutdown()
 
 
+@pytest.fixture(scope="function", autouse=True)
+def maintain_test_table(request):
+    current_step = int(request.node.name.split("[")[1].split("-")[0])
+
+    for i, node in enumerate([node1, node2]):
+        node.query("DROP TABLE IF EXISTS testdb.test_table SYNC")
+        node.query("DROP DATABASE IF EXISTS testdb")
+
+        node.query("CREATE DATABASE testdb")
+        node.query(
+            """CREATE TABLE testdb.test_table(id UInt32, val String) ENGINE = ReplicatedMergeTree('/clickhouse/test/test_table1', '{}') ORDER BY id;""".format(
+                i
+            )
+        )
+    for i, node in enumerate([node3, node4]):
+        node.query("DROP TABLE IF EXISTS testdb.test_table SYNC")
+        node.query("DROP DATABASE IF EXISTS testdb")
+
+        node.query("CREATE DATABASE testdb")
+        node.query(
+            """CREATE TABLE testdb.test_table(id UInt32, val String) ENGINE = ReplicatedMergeTree('/clickhouse/test/test_table2', '{}') ORDER BY id;""".format(
+                i
+            )
+        )
+    yield
```

Comment:
Shouldn't it be like this?

```suggestion
    yield
    for i, node in enumerate([node3, node4]):
        node.query("DROP TABLE IF EXISTS testdb.test_table SYNC")
        node.query("DROP DATABASE IF EXISTS testdb")

        node.query("CREATE DATABASE testdb")
        node.query(
            """CREATE TABLE testdb.test_table(id UInt32, val String) ENGINE = ReplicatedMergeTree('/clickhouse/test/test_table2', '{}') ORDER BY id;""".format(
                i
            )
        )
```

Reply:
Nope.
It seems Ok to use just 'return' here , because cleanup part is empty, though 'yield' is recommended https://docs.pytest.org/en/stable/how-to/fixtures.html#yield-fixtures-recommended so I'd rather keep it as is.